examples/advection_1d/advection_1d_nonunif.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+r"""
+One-dimensional advection
+=========================
+
+Solve the linear advection equation:
+
+.. math::
+    q_t + u q_x = 0.
+
+Here q is the density of some conserved quantity and u is the velocity.
+Here we have a nonuniform grid, given by the transformation x**2 on grid [-0.5,0.5] to [-0.25,0.25].
+The initial condition is a Gaussian centered at 0 and the boundary conditions are periodic.
+The final solution is identical to the initial data because the wave has
+crossed the domain exactly once, which takes computational time 0.5, because the speed is 1 and grid length 0.5.
+"""
+from __future__ import absolute_import
+import numpy as np
+from clawpack import riemann
+
+def mapc2p_nonunif(xc):
+    """This function takes the interval [-xL,xR] and squares the computational coordinate
+       while keeping the negative coordinates to be squared and yet retain their negativity
+       to the physical coordinate [-xL**2, xR**2]
+       This serves as an example of transformation to a nonuniform grid.
+    """
+    neg = -1*(xc < 0) + (xc > 0)
+    xp = xc**2
+    xp = neg*xp
+    return xp
+
+
+
+def setup(nx=100, kernel_language='Python', use_petsc=False, solver_type='classic', weno_order=5,
+          time_integrator='SSP104', outdir='./_output'):
+
+    if use_petsc:
+        import clawpack.petclaw as pyclaw
+    from clawpack import pyclaw
+    import clawpack.pyclaw.geometry
+
+    if kernel_language == 'Fortran':
+        riemann_solver = riemann.advection_1D
+    elif kernel_language == 'Python':
+        riemann_solver = riemann.advection_1D_py.advection_1D
+
+    if solver_type=='classic':
+        solver = pyclaw.ClawSolver1D(riemann_solver)
+    elif solver_type=='sharpclaw':
+        solver = pyclaw.SharpClawSolver1D(riemann_solver)
+        solver.weno_order = weno_order
+        solver.time_integrator = time_integrator
+        if time_integrator == 'SSPLMMk3':
+            solver.lmm_steps = 5
+            solver.check_lmm_cond = True
+    else: raise Exception('Unrecognized value of solver_type.')
+
+    solver.kernel_language = kernel_language
+    #solver.order = 2 # in order to make test file pass without complaint
```

Comment:
It does not work on `order=2`?  That seems odd.

Reply:
it works on order 2, but when Travis runs the tests, if the solver.order is set to 1 or 2 or anything, it gives an error saying that SharpClaw solver does not have order as an attribute and the whole test stalls.